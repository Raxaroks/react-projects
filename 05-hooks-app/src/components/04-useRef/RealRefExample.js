import React, { useState} from 'react';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';
import '../03-examples/styles.css';

const RealRefExample = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1><strong>RealRefExample</strong></h1> <hr/>

            {
                show && <MultipleCustomHooks/>
            }

            <button
                className='btn btn-primary mt-4'
                onClick={ () => { setShow(!show) }}
            >
                Show/Hide
            </button>
        </div>
    );
}

export default RealRefExample;
