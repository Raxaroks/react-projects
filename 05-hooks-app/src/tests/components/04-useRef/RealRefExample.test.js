import '@testing-library/jest-dom';
import '@testing-library/react-hooks';
import { shallow } from 'enzyme';
import RealRefExample from '../../../components/04-useRef/RealRefExample';

describe('Pruebas en el componente: RealRefExample', () => {
    
    const componentWrapper = shallow(<RealRefExample/>);

    test('Debe de renderizarse correctamente.', () => {
        expect(componentWrapper).toMatchSnapshot();
        expect( componentWrapper.find('MultipleCustomHooks').exists() ).toBe(false);
    });    

    test('Debe de mostrarse correctamente el componente <MultipleCustomHooks/>', () => {
        componentWrapper.find('button').simulate('click');
        expect( componentWrapper.find('MultipleCustomHooks').exists() ).toBe(true);        
    });  

});

