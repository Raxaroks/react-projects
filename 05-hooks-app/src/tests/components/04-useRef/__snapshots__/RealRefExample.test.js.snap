// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente: RealRefExample Debe de renderizarse correctamente. 1`] = `
<div>
  <h1>
    <strong>
      RealRefExample
    </strong>
  </h1>
   
  <hr />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
